recipes-bsp/uefi/tegra-uefi-capsules_35.5.0.bb

```diff
@@ -2,54 +2,67 @@ DESCRIPTION = "Generate UEFI capsules for bup paylods"
 LICENSE = "MIT"
 LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
 
-inherit tegra-bup deploy
+inherit tegra-bup deploy image_types_tegra kernel-artifact-names
 
 TEGRA_UEFI_CAPSULE_SIGNING_CLASS ??= "tegra-uefi-capsule-signing"
 inherit ${TEGRA_UEFI_CAPSULE_SIGNING_CLASS}
 
 TEGRA_UEFI_CAPSULE_SIGNING_EXTRA_DEPS ??= ""
 
 COMPATIBLE_MACHINE = "(tegra)"
-
-DEPENDS += "tegra-bup-payload"
+TEGRA_SIGNING_EXTRA_DEPS ??= ""
 
 GUID:tegra194 ?= "be3f5d68-7654-4ed2-838c-2a2faf901a78"
 GUID:tegra234 ?= "bf0d4599-20d4-414e-b2c5-3595b1cda402"
 
 do_compile() {
-    sign_uefi_capsules
+
+    if [ ! -z "${INITRAMFS_IMAGE}" ]; then
+        initramfs_symlink_name=${INITRAMFS_IMAGE}-${MACHINE}
+        # Generate BUP images
+        oe_make_bup_payload ${DEPLOY_DIR_IMAGE}/${initramfs_symlink_name}.cboot
+
+        # Generate UEFI capsules
+        generate_uefi_capsule ${WORKDIR}/bup-payload/bl_only_payload tegra-bl.cap
+        generate_uefi_capsule ${WORKDIR}/bup-payload/kernel_only_payload tegra-kernel.cap
```

Comment:
This should be conditional on the existence of the kernel-only payload.  It's technically possible to remove the kernel and DTB partitions from a storage layout, in which case no such payload would get generated.

Reply:
My understanding is that the kernel payload as a capsule doesn't work to start with. If this is correct, it might make sense to omit it altogether.